src/Nethermind/Nethermind.Optimism.Test/OptimismPayloadPreparationServiceTests.cs

```diff
@@ -1,42 +1,54 @@
 // SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-using Nethermind.Core.Test.Builders;
-using NUnit.Framework;
-using FluentAssertions;
+using System.Threading.Tasks;
+using System.Collections.Generic;
 using System;
+using NUnit.Framework;
 using NSubstitute;
-using Nethermind.Core.Specs;
+using Nethermind.State;
+using Nethermind.Optimism.Rpc;
 using Nethermind.Merge.Plugin.BlockProduction;
-using Nethermind.Core.Timers;
 using Nethermind.Logging;
-using Nethermind.Optimism.Rpc;
+using Nethermind.Int256;
+using Nethermind.Evm.Tracing;
+using Nethermind.Core.Timers;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Core.Specs;
+using Nethermind.Core.Crypto;
+using Nethermind.Core;
 using Nethermind.Consensus.Transactions;
 using Nethermind.Consensus.Processing;
-using Nethermind.Blockchain;
-using Nethermind.State;
 using Nethermind.Consensus;
-using Nethermind.Core;
 using Nethermind.Config;
-using Nethermind.Core.Crypto;
-using Nethermind.Evm.Tracing;
-using System.Buffers.Binary;
-using System.Threading.Tasks;
+using Nethermind.Blockchain;
+using FluentAssertions;
+using Nethermind.Crypto;
 
 namespace Nethermind.Optimism.Test;
 
 [Parallelizable(ParallelScope.All)]
 public class OptimismPayloadPreparationServiceTests
 {
-    [TestCase(8u, 2u)]
-    [TestCase(2u, 2u)]
-    [TestCase(2u, 10u)]
-    public async Task Writes_EIP1559Params_Into_HeaderExtraData(UInt32 denominator, UInt32 elasticity)
+    private static IEnumerable<(OptimismPayloadAttributes, EIP1559Parameters?)> TestCases()
+    {
+        foreach (var noTxPool in (bool[])[true, false])
+        {
+            yield return (new OptimismPayloadAttributes { EIP1559Params = [0, 0, 0, 8, 0, 0, 0, 2], NoTxPool = noTxPool }, new EIP1559Parameters(0, 8, 2));
+            yield return (new OptimismPayloadAttributes { EIP1559Params = [0, 0, 0, 2, 0, 0, 0, 2], NoTxPool = noTxPool }, new EIP1559Parameters(0, 2, 2));
+            yield return (new OptimismPayloadAttributes { EIP1559Params = [0, 0, 0, 2, 0, 0, 0, 10], NoTxPool = noTxPool }, new EIP1559Parameters(0, 2, 10));
+            yield return (new OptimismPayloadAttributes { EIP1559Params = [0, 0, 0, 0, 0, 0, 0, 0], NoTxPool = noTxPool }, new EIP1559Parameters(0, 250, 6));
+        }
```

Comment:
We added tests that set the `NoTxPool = true`, which showed that the `PayloadPreparationService` was not adjusting the `BlockHeader.ExtraData` with the EIP-1559 parameters even though this should always be done.